### DISCUSSION TITLE
Button type

### DISCUSSION THREAD

**Discussion by inezabonte** [vercel/next.js#24705] [Help]
Is it necessary to give the button a type="button" in Next.js
Cc: @Ryuno-ki

**Comment by bobarros** [comment]
If this is a question, the answer is no.

The example below is perfectly fine.

```html
<button>submit</button>
```

**Reply by Ryuno-Ki** [reply]
Thanks for tagging.
I disagree with @bobarros.
See https://lea.verou.me/2018/05/never-forget-typebutton-on-generated-buttons/ for motivation.

The default value of the `type` of a `<button>` is `submit`. That's not always preferable.
Thus, the option to define the `type` is needed.

**Comment by bobarros** [comment]
It's not about agreeing or disagreeing. The word "necessary" means something.
You have a nice article about people forgetting that button has "submit" as default.

Just look at the last phrase of the text, though:
_______
_Most of the time a type-less <button> will work just fine, but when it doesn’t, things get really weird._
______

Does this look like something necessary in all cases? And when you actually need a submit behavior? It's a good practice to have type="submit"? Maybe. Necessary, also no.

It seems that you guys brought a previous discussion to this board and expected others to be on the same page.

**Reply by Ryuno-Ki** [reply]
Actually, yes. https://github.com/inezabonte/inezabonte/issues/70 sparked this topic.
